File: LightOJ/Code Samurai Preliminary/aa.cpp
#include <iostream>
#include <string.h>
using namespace std;
int main() {
    int t; cin >> t;
    while(t--) {
        int n; cin >> n;
        char m[26] = {0};

        for(int i = 0; i < n; i++) {
            char s[10]; cin >> s;
            m[s[0] - 'a']++;
        }

        char ans[] = "impossible";

        for(int i = 0; i < 26; i++) {
            if(m[i] == 3) {
                ans[0] = 'a' + i; break;
            }
        }

        cout << ans << endl;

    }

    return 0;
}

File: LightOJ/Code Samurai Preliminary/aa.py
from collections import defaultdict


for _ in range(int(input())):
	n = int(input())

	m = defaultdict(int)
	names = []
	for i in range(n):
		names.append(input())
		

	ans = "impossible"
	for n in names:
		m[n[0]]+=1

		if m[n[0]] == 3:
			ans = n[0]
			break


	print(ans)